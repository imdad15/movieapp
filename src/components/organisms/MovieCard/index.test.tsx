import React from "react";
import { render, screen } from "@testing-library/react";
import { MovieCard } from ".";
import { useNavigate } from "react-router-dom";

jest.mock("react-router-dom", () => ({
  useNavigate: jest.fn(),
}));

describe("Search icon tests", () => {
  beforeEach(() => {
    useNavigate as jest.Mock;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should match snapshot", () => {
    const { container: component } = render(
      <MovieCard id={"asc123"} title={"New Movie"} poster={"PosterUrl"} />
    );
    expect(component).toMatchSnapshot();
  });

  it("should render the Movie card component with props", () => {
    render(
      <MovieCard id={"asc123"} title={"New Movie"} poster={"PosterUrl"} />
    );
    expect(screen.getByText("New Movie")).toBeDefined();
  });
});
