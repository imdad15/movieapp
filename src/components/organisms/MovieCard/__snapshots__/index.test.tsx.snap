// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search icon tests should match snapshot 1`] = `
<div>
  <div
    class="lazyload-wrapper "
  >
    <div
      class="lazyload-placeholder"
      style="height: 200px;"
    />
  </div>
</div>
`;
